torch/nn/modules/flatten.py

```diff
@@ -57,8 +58,10 @@ class Unflatten(Module):
       (tuple of `(name, size)` tuples) for `NamedTensor` input.
 
     Shape:
-        - Input: :math:`(N, *dims)`
-        - Output: :math:`(N, C_{\text{out}}, H_{\text{out}}, W_{\text{out}})`
+        - Input: :math:`(*, S_{dim}, *)`. Input can be any shape, where :math:`S_{dim}`
+          is the size at dimension :math:`dim`, :math:`S_{dim}=\prod *unflattened\_size`, and
```

Comment:
I realized technically `\prod_i unflattened\_size[i]` is probably more accurate than `\prod *unflattened\_size`.

Alternatively, maybe a `U_{i}` notation for the unflattened size?